implementations/elixir/ockam/ockam/lib/ockam/transport/portal/msgs.ex

```diff
@@ -0,0 +1,17 @@
+defmodule Ockam.Transport.Portal.TunnelProtocol do
+    @typedoc """
+    Tunneling protocol messages
+    """
+    alias Ockam.Bare.Extended, as: BareExt
+    @type msg :: :ping | :pong | :disconnect | {:payload, binary()}
+
+    # a bit redundant, likely this can be inferred from the msg() type above via macro?
+    @schema  {:variant, [:ping, :pong, :disconnect, {:payload, :data}]}
```

Comment:
Personally I prefer explicit schemas when dealing with cross-implementation protocols.
I like the approach of defining a module per protocol message though, we'll need to do that in other protocols as well.